lib/jiff-client.js

```diff
@@ -4420,21 +4673,240 @@
+     * @method get_preprocessing
+     * @memberof jiff-instance
+     * @instance
+     * @param {string} op_id - the op_id associated with the preprocessed value/share.
+     */
+    jiff.get_preprocessing = function (op_id) {
+      var values = jiff.preprocessing_table[op_id];
+      if (values == null) {
+        throw new Error('No preprocessed value(s) that correspond to the op_id "' + op_id + '"');
+      }
+      return values;
+
+      // TODO: do some fallback according to user.
+      // return jiff.triplet(self.holders, max(self.threshold, o.threshold), self.Zp, op_id + ':triplet');
+      // the problem w/ this is there is no good way to determine what sort of value/object you need from the op_id. parsing it for relevant text isn't possible because it may be a nested operation
+      // need a map like if ':triplet' = jiff.triplet()
+    };
+
+    /**
+     * Store a pair of op_id and associated pre-processed value/share.
+     * The value/share can be accessed later during the computation through jiff.get_preprocessing(op_id).
+     * @method store_preprocessing
+     * @memberof jiff-instance
+     * @instance
+     * @param {string} op_id - the op_id associated with the preprocessed value/share.
+     * @param {secret_share} share - the share/value to store.
+     */
+    jiff.store_preprocessing = function (op_id, share) {
+      jiff.preprocessing_table[op_id] = share;
+    };
+
+    /**
+     * Generate values used for jiff operations in advance of the general computation
+     * @method preprocessing
+     * @memberof jiff-instance
+     * @instance
+     * @param {string} dependent_op - name of the operation that will later use the pre_processed values
+     * @param {Number} [count=1] - number of times the protocol should be performed, number of values that will be generated
+     * @param {Object} [protocols=defaults] - a mapping from base preprocessing elements (triplets, bit arrays) to functions that can pre-process them
+     *                               the function must implement the same interface as the JIFF provided protocols (e.g. jiff.protocols.generate_beaver_bgw).
+     *                               missing mappings indicate that JIFF must use the default protocols.
+     * @param {Number} [threshold=receivers_list.length] - the threshold of the preprocessed shares.
+     * @param {Array} [receivers_list=all_parties] - the parties that will receive the preprocsssed shares.
+     * @param {Array} [compute_list=all_parties] - the parties that will compute the preprocsssed shares.
+     * @param {Array} [Zp=jiff.Zp] - the Zp of the preprocessed shares.
+     * @param {Array} [id_list=auto_gen()] - array of ids to be used sequentially to identify the pre_processed values. Optional.
+     * @param {Object} [params={}] - any additional protocol-specific parameters.
+     */
+    jiff.preprocessing = function (dependent_op, count, protocols, threshold, receivers_list, compute_list, Zp, id_list, params) {
```

Comment:
add if statement for invalid dependent_op (null, does not exist in protocols, does not exist in pre-processing map)